### PR TITLE
Fix Conversion instances 

### PR DESCRIPTION
**Pull request by epicallan**
* [x] Depends on #244

### CODE REVIEW

File: conversions/src/Data/Conversions.hs [hunk 1]
```diff
@@ -183,7 +183,12 @@ instance (MonadError (BoundError Int Word32) m) => Conversion (m Word32) Int whe
   convert = checkedFromIntegralToBounded
 
 instance (MonadError (BoundError Int Word64) m) => Conversion (m Word64) Int where
-  convert = checkedFromIntegralToBounded
+  convert value = do
+    when (value < 0) $ throwError error
+
+    maybe (throwError error) pure $ checkedFromIntegral value
+      where
+        error = BoundError value
```
Comment (epicallan) [inline conversions/src/Data/Conversions.hs:194]: comment: `boundError` binding is already in top level scope, I am therefore using `error`.

File: conversions/src/Data/Conversions.hs [hunk 2]
```diff
@@ -225,12 +225,12 @@ convertErrorFromNatural value =
   maybe (throwError $ UserBoundError value minBound maxBound) pure
     $ checkedFromIntegral value
 
-convertErrorFromIntegral
+convertErrorFromInt
```
Comment (epicallan) [inline conversions/src/Data/Conversions.hs:228]: Comment: I am now thinking `converErrorFromSizedInt` may be a better pick.

Comment (mbj) [reply conversions/src/Data/Conversions.hs:228]: I do not agree the current name is an improvement, as `Int` implies `Data.Int(Int)` which has nothing to do with this specific instance. What about `convertErrorBounded`?

Comment (epicallan) [reply conversions/src/Data/Conversions.hs:228]: @mbj you are right. However, we also have `convertBoundedFromIntegral` which is quite similar to `convertErrorBounded` implementation-wise with minor changes. I see no obvious course of action for now, something to think about.

**Review by mbj — APPROVED** [review]

File: conversions/src/Data/Conversions.hs [hunk 3]
```diff
@@ -225,12 +225,12 @@ convertErrorFromNatural value =
   maybe (throwError $ UserBoundError value minBound maxBound) pure
     $ checkedFromIntegral value
 
-convertErrorFromIntegral
+convertErrorBounded
```
Comment (dkubb) [inline conversions/src/Data/Conversions.hs:228]: __suggestion:__ The git commit message references the old `convertErrorFromInt` name.

**Review by dkubb — APPROVED** [review]

**Pull request merged**